docs/block-processing/us.md

```diff
@@ -0,0 +1,359 @@
+`max spMinThd (spInitThd - (t - s) * spThdDecrement)`, then proposal's `BlockVersion` becomes **adopted**.
+
+Intuitively, the threshold which needed to adopt `BlockVersion` is decreasing in each epoch but cannot become
+less than some reasonable minimal value (`spMinThd`).
+
+There is datatype which can also be updated by `BlockVersionModifier`:
+```
+data SoftforkRule = SoftforkRule
+    { srInitThd      :: !CoinPortion
+    , srMinThd       :: !CoinPortion
+    , srThdDecrement :: !CoinPortion
+    }
+```
+
+So we check this rule at the beginning of each epoch for each _confirmed_ proposal 
+and adopt one of block version from _confirmed_ proposals if this version satisfies the rule.
+
+By the way, we call block version **competing** if it might become adopted. 
+It implies there is at least one confirmed proposal with such block version.
+
+#### Block validation according to adopted version
+
+Assume a proposal which is bumping `BlockVersion` became _confirmed_.
+Though the proposal is already _confirmed_ but stakeholders
+haven't updated its software yet, hence, 
+they can't validate blocks of block version from the proposal.
+
+To avoid a situation when adversary stakeholder issues invalid block of _competing_ version
+others stakeholders will validate blocks according to the current _adopted_ block version.
+
+#### Example
+
+Assume there are four stakeholders with stakes `[0.25, 0.26, 0.1, 0.39]`,
+`lastKnownBlockVersion` is `0.1.0` for all stakeholders' softwares and 
+last adopted block version is also `0.1.0`.
+
+Then update proposal which bumps block version to `0.2.0` is confirmed in 2nd epoch.
+Suppose this proposal adds consensus rule, 
+that block header's attributes should contain by key `228` string `"pva"`.
+
+Then the first stakeholder downloads this update and update own software. 
+`lastKnownBlockVersion` is `0.2.0` for this software but last adopted block version is still `0.1.0`.
+
+Then it's turn to issue a block of the first stakeholder as leader of some slot.  
+To be valid for others stakeholders, issued block should be created by rules for last adopted version, so
+updated software of the first stakeholder considers last adopted block version.  
+If last adopted is `0.1.0` then the software creates block using old logic (logic for `0.1.0`)
+otherwise using new logic (logic for `0.2.0`).  
+The last adopted version is `0.1.0`, hence, the first stakeholder issues a block:
+* `BlockVersion` of this block is set to `0.2.0`
+* block created by rules for last adopted block version, i.e. `0.1.0`. 
+
+So the block is issued without new attribute but tagged by the new version.
+Stakeholders with old software expects that all attributes are known because
+last adopted block version equals `lastKnownBlockVersion` for its softwares.
+
+Then the 3rd epoch starts and the threshold (which is regulated by `SoftforkRule`) 
+for adopting new block version is dumped to `0.6`.
```

Comment:
I suggest that in this example we use values from mainnet, where we have initial threshold 0.9, minimal threshold 0.6 and decrement 0.05.